homeassistant/components/hydrawise/entity.py

```diff
@@ -24,21 +24,25 @@ def __init__(
         coordinator: HydrawiseDataUpdateCoordinator,
         description: EntityDescription,
         controller: Controller,
+        *,
         zone: Zone | None = None,
+        sensor: Sensor | None = None,
     ) -> None:
         """Initialize the Hydrawise entity."""
         super().__init__(coordinator=coordinator)
         self.entity_description = description
         self.controller = controller
         self.zone = zone
-        self._device_id = str(controller.id if zone is None else zone.id)
+        self.sensor = sensor
```

Comment:
Proposal:
instead make the zone_id and the sensor_id a property of the object. Then add some extra properties
```py
    @property
    def zone(self) -> Zone:
        return self.coordinator.data.zones[self.zone_id]

    @property
    def sensor(self) -> Sensor:
        return self.coordinator.data.sensors[self.sensor_id]
```

This way, you don't have to add the if statements in line 57-60

Reply:
Fixed.